### PR TITLE
[STALE] Issue/fast http poc 

### PR DESCRIPTION
**Pull request by yonidavidson**
This is a work in progress to fix issue #152

### CODE REVIEW

**Comment by yonidavidson** [comment]
@lbeder  @talkol do you want me to do anything with this branch or just close the PR for now?

**Comment by electricmonk** [comment]
@yonidavidson how do you feel about adding a Go Bench test comparing the two? :)

File: bootstrap/httpserver/fasthttp_server.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		publicApi: publicApi,
+	}
+
+	readTimeout, _ := time.ParseDuration("500ms") //TODO error on failed parse
+	server.httpServer = &fasthttp.Server{
+		Handler:     server.createRouter(),
+		ReadTimeout: readTimeout,
+	}
+
+	go func() {
```
Comment (netoneko) [inline bootstrap/httpserver/fasthttp_server.go]: I think we should pass context here so we could stop the server gracefully.

Comment (yonidavidson) [reply bootstrap/httpserver/fasthttp_server.go]: What do you mean? the shutdown function waits for all connections to finish and then shuts down gracefully.

**Comment by yonidavidson** [comment]
Profiling for fast_http:
`Showing nodes accounting for 10ms, 100% of 10ms total
Showing top 10 nodes out of 19
      flat  flat%   sum%        cum   cum%
      10ms   100%   100%       10ms   100%  runtime.memclrNoHeapPointers
         0     0%   100%       10ms   100%  github.com/orbs-network/orbs-network-go/instrumentation/log.(*basicLogger).Info
         0     0%   100%       10ms   100%  github.com/orbs-network/orbs-network-go/instrumentation/log.(*basicLogger).Log
         0     0%   100%       10ms   100%  github.com/orbs-network/orbs-network-go/instrumentation/log.(*humanReadableFormatter).FormatRow
         0     0%   100%       10ms   100%  github.com/orbs-network/orbs-network-go/instrumentation/log.printParam
         0     0%   100%       10ms   100%  github.com/orbs-network/orbs-network-go/services/consensusalgo/benchmarkconsensus.(*service).leaderConsensusRoundRunLoop
         0     0%   100%       10ms   100%  github.com/orbs-network/orbs-network-go/services/consensusalgo/benchmarkconsensus.(*service).leaderConsensusRoundTick
         0     0%   100%       10ms   100%  github.com/orbs-network/orbs-network-go/services/consensusalgo/benchmarkconsensus.(*service).leaderGenerateNewProposedBlockUnderMutex
         0     0%   100%       10ms   100%  github.com/orbs-network/orbs-network-go/services/consensuscontext.(*service).RequestNewResultsBlock
         0     0%   100%       10ms   100%  runtime.(*mcache).nextFree
(pprof) `

**Comment by yonidavidson** [comment]
@netoneko  @electricmonk  I want to benchmark the e2e (so we'll know if the fasthttp actually creates a improvement on the system.
a. Do you think this approach is good ?  - @netoneko suggested that I'll just benchmark the server but a server without nothing we already have the benchmark.
b. The E2E tests fail due to all the flakiness , is there a way I can try force it not to fail (I can stop the benchmark counter before any hack needed for it to work)

**Comment by yonidavidson** [comment]
HttpServer:
goos: darwin
goarch: amd64
pkg: github.com/orbs-network/orbs-network-go/bootstrap/httpserver
10000	    123541 ns/op
PASS

Process finished with exit code 0

FastHttpServer:

goos: darwin

2018/09/20 23:42:39 Unsolicited response received on idle HTTP channel starting with "HTTP/1.1 400 Bad Request\r\nServer: fasthttp\r\nDate: Thu, 20 Sep 2018 20:42:39 GMT\r\nContent-Type: text/plain; charset=utf-8\r\nContent-Length: 26\r\nConnection: close\r\n\r\nError when parsing request"; err=<nil>
goarch: amd64
pkg: github.com/orbs-network/orbs-network-go/bootstrap/httpserver

2018/09/20 23:42:39 Unsolicited response received on idle HTTP channel starting with "HTTP/1.1 400 Bad Request\r\nServer: fasthttp\r\nDate: Thu, 20 Sep 2018 20:42:39 GMT\r\nContent-Type: text/plain; charset=utf-8\r\nContent-Length: 26\r\nConnection: close\r\n\r\nError when parsing request"; err=<nil>

2018/09/20 23:42:40 Unsolicited response received on idle HTTP channel starting with "HTTP/1.1 400 Bad Request\r\nServer: fasthttp\r\nDate: Thu, 20 Sep 2018 20:42:40 GMT\r\nContent-Type: text/plain; charset=utf-8\r\nContent-Length: 26\r\nConnection: close\r\n\r\nError when parsing request"; err=<nil>
10000	    101495 ns/op
PASS

Process finished with exit code 0



So even that the fasthttp server gets some warning it seems that it's faster by 18%.

**Comment by yonidavidson** [comment]
@netoneko can I close this PR? it's getting stale.

**Comment by electricmonk** [comment]
please leave it open. We can let it go stale in the meantime.

**Pull request closed without merging**